datafusion/core/src/physical_plan/file_format/mod.rs

```diff
@@ -130,7 +130,9 @@ impl FileScanConfig {
             column_statistics: Some(table_cols_stats),
         };
 
-        let table_schema = Arc::new(Schema::new(table_fields));
+        let table_schema = Arc::new(
+            Schema::new(table_fields).with_metadata(self.file_schema.metadata().clone()),
```

Comment:
Here is the change that is required to preserve the metadata in the merged schema